src/planner/binder/expression/bind_operator_expression.cpp

```diff
@@ -13,7 +13,11 @@ namespace duckdb {
 
 LogicalType ExpressionBinder::ResolveNotType(OperatorExpression &op, vector<unique_ptr<Expression>> &children) {
 	// NOT expression, cast child to BOOLEAN
-	D_ASSERT(children.size() == 1);
+	if (children.size() != 1) {
```

Comment:
I think this is the wrong place to fix the error - NOT operators should not be generated with multiple expressions. I think we should be throwing this error when expanding the `*COLUMNS(*)` expression instead